runtime/src/current_sandbox.c

```diff
@@ -88,20 +103,43 @@ current_sandbox_start(void)
 	current_sandbox_disable_preemption(sandbox);
 	sandbox->completion_timestamp = __getcycles();
 
-	/* Retrieve the result, construct the HTTP response, and send to client */
-	if (sandbox_send_response(sandbox) < 0) {
-		error_message = "Unable to build and send client response\n";
-		goto err;
-	};
 
-	http_total_increment_2xx();
+	if (next_module != NULL) {
+		/* Generate a new request, copy the current sandbox's output to the next request's buffer, and put it to the global queue */
+		ssize_t output_length = sandbox->request_response_data_length - sandbox->request_length;
+		char * pre_func_output = (char *) malloc(output_length);
+		memcpy(pre_func_output, sandbox->request_response_data + sandbox->request_length, output_length);
+	 	uint64_t enqueue_timestamp = __getcycles();	
+		struct sandbox_request *sandbox_request =
+                                  sandbox_request_allocate(next_module, false, sandbox->request_length, 
+							   next_module->name, sandbox->client_socket_descriptor,
+                                                           (const struct sockaddr *)&sandbox->client_address,
+                                                           sandbox->request_arrival_timestamp, enqueue_timestamp, 
+							   true, pre_func_output, output_length);
+		/* TODO: all sandboxs in the chain share the same request id, but sandbox_request_allocate() will busy-wait to generate an unique
+		   id, should we optimize it here?*/
```

Comment:
> It uses `faa` to retrieve the new id, right? In that case, there should be no busy waiting.
> 
> If it doesn't use `faa`, we should certainly discuss optimizations.
> 
> Separately, it would be good to get rid of the `faa` on global memory. Easy enough to have the sandbox ids be retrieved with a `faa` on local (`__thread`) memory bitwise appended onto the worker id. @bushidocodes Is this insane? Should we create a card? Certainly, this is very low priority.

what is `faa` ? sledge get this id by call C function `atomic_fetch_add`, it applies for the variable defined by `_Atomic` keyword. I am not sure the compiler will use the software or hardware to guarantee the atomic operation.